Source/WebInspectorUI/UserInterface/Controllers/DeviceSettingsManager.js

```diff
@@ -0,0 +1,178 @@
+    get overridenDeviceUserAgent() { return this._overridenDeviceUserAgent; }
+    get overridenDeviceScreenSize() { return this._overridenDeviceScreenSize; }
+    get overridenDeviceSettings() { return this._overridenDeviceSettings; }
+
+    get hasOverridenDefaultSettings() {
+        return this._overridenDeviceUserAgent || this._overridenDeviceScreenSize || this._overridenDeviceSettings.size > 0;
+    }
+
+    overrideDeviceSetting(setting, value, callback)
+    {
+        let target = WI.assumingMainTarget();
+        if (!target.hasCommand("Page.overrideSetting")) {
+            callback(false);
+            return;
+        }
+
+        let commandArguments = {
+            setting,
+        };
+
+        let shouldOverride = !this._overridenDeviceSettings.has(setting)
+            commandArguments.value = value; // Providing `value` applies the override for `setting`. Not providing it clears the override.
+
+        target.PageAgent.overrideSetting.invoke(commandArguments, (error) => {
+            if (error) {
+                WI.reportInternalError(error);
+                callback(false);
+                return;
+            }
+
+            if (shouldOverride)
+                this._overridenDeviceSettings.set(setting, value);
+            else
+                this._overridenDeviceSettings.delete(setting);
+
+            callback(shouldOverride);
+            this.dispatchEventToListeners(WI.DeviceSettingsManager.Event.SettingChanged);
+        });
+    }
+
+    overrideUserAgent(value, force)
+    {
+        if (value === this._overridenDeviceUserAgent)
+            return;
+
+        let target = WI.assumingMainTarget();
+        if (!target.hasCommand("Page.overrideUserAgent"))
+            return;
+
+        let commandArguments = {};
+
+        let shouldOverride = value && (value !== WI.DeviceSettingsManager.DefaultValue || force);
+        if (shouldOverride)
+            commandArguments.value = value; // Providing `value` applies the user agent string override. Not providing it clears the override.
+
+        target.PageAgent.overrideUserAgent.invoke(commandArguments, (error) => {
+            if (error) {
+                WI.reportInternalError(error);
+                return;
+            }
+
+            this._overridenDeviceUserAgent = shouldOverride ? value : null;
+
+            this.dispatchEventToListeners(WI.DeviceSettingsManager.Event.SettingChanged);
+            target.PageAgent.reload();
+        });
+    }
+
+    overrideScreenSize(value, force)
+    {
+        if (value === this._overridenDeviceScreenSize)
+            return;
+
+        let target = WI.assumingMainTarget();
+        if (!target.hasCommand("Page.setScreenSizeOverride"))
+            return;
+
+        let commandArguments = {};
+
+        let shouldOverride = value && (value !== WI.DeviceSettingsManager.DefaultValue || force);
+        if (shouldOverride) {
+            let tokens = value.split("x");
```

Comment:
NIT: you could `let [width, height] = value.split("x");` instead of grabbing them later

Reply:
Neat!